engine/flink/components/dev-table/src/main/scala/pl/touk/nussknacker/engine/flink/table/sink/SqlSinkFactory.scala

```diff
@@ -0,0 +1,74 @@
+  NodeDependencyValue,
+  SingleInputDynamicComponent
+}
+import pl.touk.nussknacker.engine.api.definition.{NodeDependency, ParameterWithExtractor, TypedNodeDependency}
+import pl.touk.nussknacker.engine.api.process.{Sink, SinkFactory}
+import pl.touk.nussknacker.engine.api.{NodeId, Params}
+import pl.touk.nussknacker.engine.flink.table.sink.SqlSinkFactory.ValueParamName
+import pl.touk.nussknacker.engine.flink.table.utils.SqlComponentFactory
+import pl.touk.nussknacker.engine.flink.table.utils.SqlComponentFactory.getSelectedTableUnsafe
+import pl.touk.nussknacker.engine.flink.table.{SqlDataSourcesDefinition, TableDefinition}
+import pl.touk.nussknacker.engine.util.parameters.SingleSchemaBasedParameter
+
+object SqlSinkFactory {
+  val ValueParamName = "Value"
+}
+
+class SqlSinkFactory(definition: SqlDataSourcesDefinition) extends SingleInputDynamicComponent[Sink] with SinkFactory {
+
+  override type State = TableDefinition
+
+  // TODO: add non-raw value parameters
+  private val RawValueParam = ParameterWithExtractor.lazyMandatory[AnyRef](ValueParamName)
+
+  private val tableNameParam: ParameterWithExtractor[String] =
+    SqlComponentFactory.buildTableNameParam(definition.tableDefinitions)
+  private val TableNameParamName = tableNameParam.parameter.name
+
+  override def contextTransformation(context: ValidationContext, dependencies: List[NodeDependencyValue])(
+      implicit nodeId: NodeId
+  ): this.ContextTransformationDefinition = {
+    case TransformationStep(Nil, _) =>
+      NextParameters(
+        parameters = RawValueParam.parameter :: tableNameParam.parameter :: Nil,
+        errors = List.empty,
+        state = None
+      )
+    case TransformationStep(
+          (ValueParamName, rawValueParamValue) ::
+          (TableNameParamName, DefinedEagerParameter(tableName: String, _)) :: Nil,
+          _
+        ) =>
+      val selectedTable = getSelectedTableUnsafe(tableName, definition.tableDefinitions)
+
+      val valueParameterTypeErrors = SingleSchemaBasedParameter(
+        RawValueParam.parameter,
+        TypingResultOutputValidator.validate(_, selectedTable.typingResult)
+      ).validateParams(Map(ValueParamName -> rawValueParamValue)).fold(_.toList, _ => List.empty)
+
+      FinalResults(context, valueParameterTypeErrors, Some(selectedTable))
+  }
+
+  override def implementation(
+      params: Params,
+      dependencies: List[NodeDependencyValue],
+      finalStateOpt: Option[State]
+  ): Sink = {
+    val lazyValueParam = RawValueParam.extractValue(params)
+    val selectedTable = finalStateOpt.getOrElse(
+      throw new IllegalStateException("Unexpected (not defined) final state determined during parameters validation")
+    )
+    new SqlSink(selectedTable, definition.sqlStatements, lazyValueParam)
+  }
+
+  override def nodeDependencies: List[NodeDependency] = List(TypedNodeDependency[NodeId])
+
+  override val allowedProcessingModes: Option[Set[ProcessingMode]] = Some(Set(ProcessingMode.UnboundedStream))
```

Comment:
Let's switch to BoundedStream in source and here we can leave the default - all processing modes.

Reply:
I've changed both to allow all processing modes, left a todo and will continue this in https://github.com/TouK/nussknacker/pull/5744